### PR TITLE
T/600 Introduce static method `Selection.createFromRanges`

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Changed: Introduced static method `Selection.createFromRanges. Closes https://github.com/ckeditor/ckeditor5-engine/issues/600.

### CODE REVIEW

**Comment by ma2ciek** [comment]
Please, read https://github.com/ckeditor/ckeditor5-engine/issues/600#issuecomment-292984632.

**Comment by Reinmar** [comment]
https://github.com/ckeditor/ckeditor5-engine/issues/600#issuecomment-292988218 Sorry for that, but I'd move this to the constructor.

File: src/model/selection.js [hunk 1]
```diff
@@ -621,6 +629,21 @@ export default class Selection {
...
 
 	/**
+	 * Creates and returns an instance of `Selection` from the given ranges.
+	 * Accepts a flag describing in which way the selection is made (see {@link #addRange addRange}).
+	 *
+	 * @param {Array.<module:engine/view/range~Range>} ranges Array of ranges to set.
+	 * @param {Boolean} [isLastBackward] Flag describing if last added range was selected forward - from start to end
+	 * (`false`) or backward - from end to start (`true`). Defaults to `false`.
+	 */
+	static createFromRanges( ranges, isLastBackward ) {
```
Comment (Reinmar) [inline src/model/selection.js]: This method duplicates the constructor now so it can be removed.

**Comment by Reinmar** [comment]
> I pushed the corrected version which still uses setRanges(). But now setRanges is reassigning _ranges and _lastRangeBackward and fires change event which is unnecessary. Maybe we can use the default parameters in the constructor function + simple range instanceof checks at the end?

IMO, the code is fine now. It could be more optimised, but then it'd be harder to read. Plus, it's a premature optimisation because this way of creating a selection isn't used even once yet and you'll never be able to listen to the selection's initial `change` event too.

So, KISS. Let's save our time for more important issues.

File: src/model/selection.js (hunk 1 above)
Comment (ma2ciek) [inline src/model/selection.js]: Fixed

File: src/model/selection.js [hunk 2]
```diff
@@ -7,14 +7,14 @@
  * @module engine/model/selection
  */
 
-import Position from './position';
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
```
Comment (Reinmar) [inline src/model/selection.js]: Why did you changed the imports? Please revert this change unless it's a must-have in this ticket.
